src/com/google/javascript/jscomp/js/util/tointeger.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Converts argument to an integral numeric value.
+ *
+ * @see https://www.ecma-international.org/ecma-262/8.0/#sec-tointeger
+ *
+ * @param {*} arg
+ * @return {number}
+ */
+$jscomp.toInteger = function(arg) {
```

Comment:
I don't think this is necessary.  It should be sufficient to just write `| 0` for signed integer conversions, which doesn't need a separate file.

Reply:
The difference with `|0` is about `Infinity`:
```js
toInteger(Infinity) // Infinity
Infinity|0 // 0
```

It makes `copyWithin` unexpected behavior.
For example, `[0, 1, 2, 3].copyWithin(0, 1, Infinity)` should return `[1, 2, 3, 3]`, but it returns `[0, 1, 2, 3]` using `|0` instead of `toInteger`.

https://github.com/tc39/test262/blob/f77b930e6720f8005d5302acaedd434e13bf2fe2/test/built-ins/Array/prototype/copyWithin/non-negative-out-of-bounds-end.js#L44-L49

I've separated this because `ToInteger` is defined in [ES spec](https://www.ecma-international.org/ecma-262/8.0/#sec-tointeger) and it may be used in another polyfills.
But it is not shared currently, so I will merge the `util/tointeger.js` into `es6/array/copywithin.js`, OK?

Reply:
Merged into one file and fixed a bit to reduced the code size

Reply:
Do you know if there are other polyfills right now that would benefit from this?  If so, please feel free to leave it separate.

Reply:
Also, are we concerned with larger-than-32-bit numbers?  Reading the spec indicates that ToInteger is size-independent, so `ToInteger(2 ** 40)` should be the identity, but if we implement with `|0` then we get zero.  Probably worth a test case - in practice we'll never see such a large array (>4GB if entries were a single byte, which they're certainly not), but it's similar to handling Infinity correctly - depending where it's placed it should be the same as Infinity, or the array length, not zero.

Also, what about an array-like: {length: 2**40, [2**39]: ...} - if the number of elements to copy is small, it should still work.

I don't know of a cleverer way to do this, then.

Reply:
SGTM